lib/server_configfile.ml

```diff
@@ -1,19 +1,15 @@
 open Lwt.Syntax
 
-type build_with =
-  | Opam
-  | Dune
-
 let yaml_of_build_with = function
-  | Opam -> `String "opam"
-  | Dune -> `String "dune"
+  | Intf.Build_with.Opam -> `String "opam"
+  | Intf.Build_with.Dune -> `String "dune"
 
 let config_parser_fail value =
   failwith @@ Printf.sprintf "Config parser: value %S is invalid" value
 
 let build_with_of_yaml_exn = function
-  | `String "opam" -> Opam
-  | `String "dune" -> Dune
+  | `String "opam" -> Intf.Build_with.Opam
+  | `String "dune" -> Intf.Build_with.Dune
```

Comment:
Same question.

Reply:
Likewise same answer. The YAML code seems to be fairly contained, spreading it all over the place is probably not good, unless we move it all to the respective types.